src/components/Form.js

```diff
@@ -1,23 +1,55 @@
 import './Form.css';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { addBook } from '../redux/books/books';
 
-const Form = () => (
-  <section className="form">
-    <h3>ADD NEW BOOK</h3>
-    <form>
-      <input type="text" placeholder="Book title" required />
-      <select>
-        <option disabled selected>
-          Category
-        </option>
-        <option value="action">Action</option>
-        <option value="comedy">Comedy</option>
-        <option value="horror">Horror</option>
-        <option value="fiction">Fiction</option>
-        <option value="classics">Classics</option>
-      </select>
-      <button type="submit">ADD BOOK</button>
-    </form>
-  </section>
-);
+const Form = () => {
+  const [title, setTitle] = useState('');
+  const [category, setCategory] = useState('');
+
+  const length = useSelector(({ booksReducer }) => booksReducer.length);
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(
+      addBook({
+        id: `item${length + 1}`,
+        title,
+        category,
+      }),
+    );
+
+    setTitle('');
+    setCategory('');
+  };
+
+  return (
+    <section className="form" onSubmit={handleSubmit}>
+      <h3>ADD NEW BOOK</h3>
+      <form>
+        <input
+          type="text"
+          placeholder="Book title"
+          required
+          value={title}
+          onChange={(e) => setTitle(e.target.value)}
+        />
+        <select value={category} onChange={(e) => setCategory(e.target.value)}>
+          <option value="" disabled selected>
```

Comment:
- [ ] **[OPTIONAL]** I would recommend you resolve this warning in your console to keep with conventional code.

  ![rachid-select](https://user-images.githubusercontent.com/36542986/155124181-9a7220da-0e66-4be9-92f9-baac4adf91e2.jpg)